client/src/modules/cash/dashboard/dashboard.chart.js

```diff
@@ -0,0 +1,309 @@
+    var out = { labels : [], data : [] };
+    if (!dataset[field]) { return out; }
+
+    dataset[field].forEach(function (item) {
+      out.labels.push(item[label]);
+      out.data.push(item.count);
+    });
+
+    return out;
+  }
+
+  /**
+   * @function uniformize data
+   * @param {object} data - { data: [], labels: [] }
+   * @param {object} reversedData - { data: [], labels: [] }
+   * @param {string} periodGroup - string in ('hour', 'day', 'month')
+   */
+  function uniformize(data, reversedData, periodGroup) {
+    var hours = [
+      '01H', '02H', '03H', '04H', '05H', '06H', '07H', '08H', '09H', '10H', '11H', '12H',
+      '13H', '14H', '15H', '16H', '17H', '18H', '19H', '20H', '21H', '22H', '23H', '24H',
+    ];
+    var dayofweek = [1, 2, 3, 4, 5, 6, 7];
+    var months = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12];
+    var labels;
+    var period = {
+      hour  : hours,
+      day   : dayofweek,
+      month : months,
+    };
+
+    var base = period[periodGroup || 'hour'];
+    labels = base.concat(data.labels, reversedData.labels);
+    labels = uniqueSortedArray(labels);
+
+    return {
+      dataset : fillArray(labels, data, 'labels'),
+      reversedDataset : fillArray(labels, reversedData, 'labels'),
+    };
+  }
+
+  /** unique array */
+  function uniqueSortedArray(array) {
+    var unique;
+    var i;
+
+    if (!array.length) { return array; }
+
+    array = typeof (array[0]) === 'number' ? array.sort(sortNumber) : array.sort();
+    unique = [array[0]];
+    for (i = 1; i < array.length; i++) {
+      if (array[i] === array[i - 1]) { continue; }
+      unique.push(array[i]);
+    }
+    return unique;
+  }
+
+  /** sort number */
+  function sortNumber(a, b) {
+    return a - b;
+  }
+
+  /**
+   * @function fillArray
+   * fill an array with zero and replace elements in a correct position
+   * according the base array
+   * @param {array} base
+   * @param {object} dataset
+   * @param {string} label
+   */
+  function fillArray(base, dataset) {
+    if (base.length < dataset.labels.length) { return dataset; }
+
+    var newDataset = { data : [], labels : [] };
+
+    for (var i = 0; i < base.length; i++) {
+      newDataset.data[i] = 0;
+      newDataset.labels[i] = base[i];
+      for (var j = 0; j < dataset.labels.length; j++) {
+        if (base[i] === dataset.labels[j]) {
+          newDataset.data[i] = dataset.data[j];
+        }
+      }
+    }
+    return newDataset;
+  }
+
+  /**
+   * @method paymentSummary
+   * @description process useful payment data as summary
+   * @param {object} dataset
+   */
+  function paymentSummary(dataset) {
+    vm.summary.total = getTotals(dataset, 'countDay');
+    vm.summary.reversed = getTotals(dataset, 'reversedDay');
+    vm.summary.value = getTotals(dataset, 'value', 'value');
+  }
+
+  /** total payments */
+  function getTotals(datasets, field, valueProperty) {
+    // sum is based on day data because it doesn't have a lot of data (7 max)
```

Comment:
I was just curious because I didn't see any variables in the `getTotals` function named `day`.  If you only ever sum on days, it would be helpful to name the variables that way.